### DISCUSSION TITLE
feat: add multiple labels on elements

### DISCUSSION THREAD

**Discussion by vanfrancoisle** [clientIO/joint#2096] [Ideas]
Looking at the documentation, I noticed that there is no possibility to add multiple labels on elements like for arcs. I wonder if it would be possible to add multiple labels for elements in the library. The functions will be the same as for the arcs.
The labels are composed of a g element containing a rectangle and a text.
- Link to the source code of the arc label functions : https://github.com/clientIO/joint/blob/77712052594632377b2e7124241f017abee3c943/src/dia/Link.mjs#L254
- Link to the source code of an element : https://github.com/clientIO/joint/blob/77712052594632377b2e7124241f017abee3c943/src/dia/Element.mjs

Here is an image representing the functionality :
![Capture d’écran 2023-03-24 à 14 50 29](https://user-images.githubusercontent.com/73991929/227539441-e39ba709-f765-4fbc-a0e3-266f2a498a3e.png)

If this request is not possible, a solution is described in this url : https://groups.google.com/g/jointjs/c/r45SIrra1IQ
and another type of solution can be found in this discussion : https://github.com/clientIO/joint/discussions/1749.

**Comment by kumilingus** [comment]
It would be possible to add an API that defines element labels separately from markup. And later implement features like label dragging. But that's a very low priority for now because you can:

**1) define multiple labels already**

Add them to the element markup:
https://jsfiddle.net/kumilingus/97dwxqv3/

**2) drag labels**

Add and embed them as new elements:
https://jsfiddle.net/kumilingus/zmb6hpy7/4/
https://jsfiddle.net/kumilingus/zmb6hpy7/

**Reply by vanfrancoisle** [reply]
Okay, thank you !
